### DISCUSSION TITLE
Editing HTML

### DISCUSSION THREAD

**Discussion by folliejester** [AzuraCast/AzuraCast#4823] [Q&A]
Hey I really want to edit the public page's html and put a video in it's background. How do I do that?

**Comment by krakenfury** [comment]
They have a doc included for customization here:
https://docs.azuracast.com/en/administration/customization/css
Note that it's sort of a road map of _where_ to customize, and not quite a _how_ to. I think you'd need to learn some CSS in order to make lots of changes.

**Reply by ItsMitchh** [reply]
Pretty much what Krakenfury said. While we don't offer the ability to directly edit the HTML page but you can always use CSS to change it's background and such. Hopefully this helped

mitch

**Reply by folliejester** [reply]
But I don't think that I'll be able to set a video background with just CSS.

**Reply by Vaalyn** [reply]
You just have to also use the `Custom JS for Public Pages` to first add the video element to the page and style that via the `Custom CSS for Public Pages`.

Here is an example video background:

## Custom CSS for Public Pages
```css
[data-theme] body.page-minimal .background-video {
	width: 100vw;
	height: 100vh;
	object-fit: cover;
	position: fixed;
	left: 0;
	right: 0;
	top: 0;
	bottom: 0;
	z-index: -1;
}
```

## Custom JS for Public Pages:
```js
let videoBackgroundElement = document.createElement('video');
videoBackgroundElement.autoplay = true;
videoBackgroundElement.loop = true;
videoBackgroundElement.muted = true;
videoBackgroundElement.poster = 'https://assets.codepen.io/6093409/river.jpg';
videoBackgroundElement.className = 'background-video';

let videoBackgroundSource = document.createElement('source');
videoBackgroundSource.src = 'https://assets.codepen.io/6093409/river.mp4';
videoBackgroundSource.type = 'video/mp4';

videoBackgroundElement.appendChild(videoBackgroundSource);

document.body.append(videoBackgroundElement);
```

**Reply by folliejester** [reply]
Worked like a charm, thank you Vaalyn!

**Reply by xyzorange2** [reply]
> You just have to also use the `Custom JS for Public Pages` to first add the video element to the page and style that via the `Custom CSS for Public Pages`.
> 
> Here is an example video background:
> 
> ## Custom CSS for Public Pages
> ```css
> [data-theme] body.page-minimal .background-video {
> 	width: 100vw;
> 	height: 100vh;
> 	object-fit: cover;
> 	position: fixed;
> 	left: 0;
> 	right: 0;
> 	top: 0;
> 	bottom: 0;
> 	z-index: -1;
> }
> ```
> 
> ## Custom JS for Public Pages:
> ```js
> let videoBackgroundElement = document.createElement('video');
> videoBackgroundElement.autoplay = true;
> videoBackgroundElement.loop = true;
> videoBackgroundElement.muted = true;
> videoBackgroundElement.poster = 'https://assets.codepen.io/6093409/river.jpg';
> videoBackgroundElement.className = 'background-video';
> 
> let videoBackgroundSource = document.createElement('source');
> videoBackgroundSource.src = 'https://assets.codepen.io/6093409/river.mp4';
> videoBackgroundSource.type = 'video/mp4';
> 
> videoBackgroundElement.appendChild(videoBackgroundSource);
> 
> document.body.append(videoBackgroundElement);
> ```

The background video is in the top corner and not the whole screen. Can you please adjust your script?

**Reply by xyzorange2** [reply]
![notfullscreen](https://github.com/user-attachments/assets/77825206-664c-4be1-8f38-10685ea01abd)

I need your help, please. The background video no longer works after the Azuracast update. What do I need to do to make the video cover the entire background again?


When I scroll, the video AND the radio interface scroll up at the same time...

> You just have to also use the `Custom JS for Public Pages` to first add the video element to the page and style that via the `Custom CSS for Public Pages`.
> 
> Here is an example video background:
> 
> ## Custom CSS for Public Pages
> ```css
> [data-theme] body.page-minimal .background-video {
> 	width: 100vw;
> 	height: 100vh;
> 	object-fit: cover;
> 	position: fixed;
> 	left: 0;
> 	right: 0;
> 	top: 0;
> 	bottom: 0;
> 	z-index: -1;
> }
> ```
> 
> ## Custom JS for Public Pages:
> ```js
> let videoBackgroundElement = document.createElement('video');
> videoBackgroundElement.autoplay = true;
> videoBackgroundElement.loop = true;
> videoBackgroundElement.muted = true;
> videoBackgroundElement.poster = 'https://assets.codepen.io/6093409/river.jpg';
> videoBackgroundElement.className = 'background-video';
> 
> let videoBackgroundSource = document.createElement('source');
> videoBackgroundSource.src = 'https://assets.codepen.io/6093409/river.mp4';
> videoBackgroundSource.type = 'video/mp4';
> 
> videoBackgroundElement.appendChild(videoBackgroundSource);
> 
> document.body.append(videoBackgroundElement);
> ```

**Reply by Vaalyn** [reply]
Replace `[data-theme]` with `[data-bs-theme]` in the CSS

In the [docs for custom CSS](https://www.azuracast.com/docs/administration/css/) we've noted the following:
> An important change in newer versions…
> 
> In version 0.19.0, we switched from an old style system to a newer version of Bootstrap, and in the process, the prefix on the commands below changed from [data-theme] to [data-bs-theme]. You should update any custom CSS you’re using accordingly.

**Reply by xyzorange2** [reply]
Thank you very much, it's working again now! I've tried everything in the last few days, but I haven't found this solution. THANK YOU VERY MUCH :-)
